x/wasm/internal/keeper/genesis_test.go

```diff
@@ -482,6 +483,42 @@ func TestImportContractWithCodeHistoryReset(t *testing.T) {
 	assert.Equal(t, expHistory, keeper.GetContractHistory(ctx, contractAddr))
 }
 
+func TestImportWithGenMsg(t *testing.T) {
```

Comment:
We'll definitely need a test of setting up staking (validator set changes) in the init messages.

As well as executing a newly created contract (and giving it a bank balance).

I'm sure you have these on your TODO, just wanted to mention them

Reply:
My test strategy here is unit based. Which means the new functionality is based upon existing components and tests so that I need to cover the changes only. 
This means concrete for the `InitGenesis` method that the `msgHandler` and `stakingKeeper` are called with the right params as well as return values handled as expected. I can verify this with a mock more efficient.
What's not covered by tests is the wiring in the app package. We don't have automated tests for this, yet. I run `contrib/local` scripts on my box to verify 

To setup staking via genesis message or to execute a newly created contract is using existing logic in the keeper. Do you think we need an additional integration test for this? What makes the genesis phase different to a normal TX execution?

Reply:
I see all the detail logic is unit tested. And it is super hard to write automatic go tests for this level of interaction.

Just saying you ran one test locally full stack is good.